go/vt/vttablet/tabletserver/query_engine.go

```diff
@@ -427,25 +427,24 @@ func (qe *QueryEngine) GetMessageStreamPlan(name string) (*TabletPlan, error) {
 }
 
 // GetConnSetting returns system settings for the connection.
-func (qe *QueryEngine) GetConnSetting(ctx context.Context, settings []string) (*pools.Setting, error) {
+func (qe *QueryEngine) GetConnSetting(ctx context.Context, settings []string) (*smartconnpool.Setting, error) {
 	span, _ := trace.NewSpan(ctx, "QueryEngine.GetConnSetting")
 	defer span.Finish()
 
-	hasher := vthash.New256()
+	var buf strings.Builder
 	for _, q := range settings {
-		_, _ = hasher.WriteString(q)
+		_, _ = buf.WriteString(q)
+		_ = buf.WriteByte(';')
 	}
 
-	var cacheKey SettingsCacheKey
-	hasher.Sum(cacheKey[:0])
-
-	connSetting, _, err := qe.settings.GetOrLoad(cacheKey, 0, func() (*pools.Setting, error) {
+	cacheKey := SettingsCacheKey(buf.String())
+	connSetting, _, err := qe.settings.GetOrLoad(cacheKey, 0, func() (*smartconnpool.Setting, error) {
```

Comment:
Yeah absolutely, that's part of the new cache implementation. It's highly concurrent and uses single-flight behind the scenes to prevent two requests from planning the same setting at the same time. One of the requests will plan and the other will block until the plan is complete, to ensure they both get the exact same value :ok_hand: